### PR TITLE
SunnyWong-frontend-React

### PR DESCRIPTION
**Pull request by SunnyDoubleU**
Coding challenge implemented with React JS. It is 90% done but still missing unit testing. Though I am not very familiar with it, I will try my best to implement it in the following days. I look forward to hearing your comments.

### CODE REVIEW

File: src/logo.svg [hunk 1]
```diff
@@ -0,0 +1 @@
....23-.12s-3.58,0-5.33.12C88.45,50.57,90.2,48.48,92,46.56Zm-22.18-12A4.82,4.82,0,0,1,72.29,34a13.11,13.11,0,0,1,5.19,1.31,39.07,39.07,0,0,1,10,6.78l1.17,1.07a98.49,98.49,0,0,0-8.16,10,98.88,98.88,0,0,0-12.65,2.06c-.44-1.94-.8-3.84-1-5.67C65.8,42,67,36.24,69.81,34.6ZM64.53,85.26a58.75,58.75,0,0,1-5.54-2c-7.1-3-11.5-6.85-11.5-10.14S51.91,66,59,63.05a57.63,57.63,0,0,1,5.44-1.94A97.93,97.93,0,0,0,69,73.25,98.72,98.72,0,0,0,64.53,85.26Zm23,19.1c-8.19,7.33-13.31,8.11-15.14,8.11a4.69,4.69,0,0,1-2.42-.58c-2.86-1.63-4-7.38-3.09-15a56.07,56.07,0,0,1,1-5.78,99.51,99.51,0,0,0,12.58,2.06,97.17,97.17,0,0,0,8.24,10.08Zm4.57-4.51c-1.8-1.92-3.57-4-5.31-6.23,1.72.08,3.47.12,5.24.12s3.58,0,5.33-.11C95.63,95.85,93.87,97.93,92.09,99.85Zm22.18,12a4.82,4.82,0,0,1-2.48.59c-1.82,0-7-.8-15.16-8.1l-1.17-1.07a98.44,98.44,0,0,0,8.15-10,97,97,0,0,0,12.66-2.06c.44,1.94.79,3.84,1,5.67C118.27,104.42,117.12,110.18,114.27,111.81Zm10.8-28.44c-1.71.7-3.52,1.35-5.44,1.93a98.54,98.54,0,0,0-4.57-12.14,98.1,98.1,0,0,0,4.49-12,58.75,58.75,0,0,1,5.54,2c7.09,3,11.5,6.85,11.49,10.14S132.17,80.42,125.07,83.37ZM92,82.39a9.18,9.18,0,1,0-9.17-9.19A9.17,9.17,0,0,0,92,82.39Zm-69.32,54.5V26.2L17.89,31V141.66H139.94l4.78-4.77Z"/></svg>
```
Comment (haishanh) [inline src/logo.svg:1]: nice logo! but I think you should remove it.

File: src/pages/Overall.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    <div className="Overall-White-Container">
+                        <div className="filter">
+                            <div className="input-class">
+                                <select class="custom-select" style={{ width: "200px" }} required value={this.state.filterType} onChange={this.changeHandler}>
+                                    <option className="select-items" defaultValue>Filter</option>
+                                    <option value="region">Regions</option>
+                                    <option value="lastinput">Last Input</option>
+                                    <option value="formnumbers">Form Numbers</option>
+                                    <option value="voternumbers">Voter Numbers</option>
+                                    <option value="update">Update</option>
```
Comment (haishanh) [inline src/pages/Overall.js:79]: why don't just define theses options as an array variable and call `map` on the array here.

File: src/pages/Overall.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                                </select>
+                            </div>
+                            <input className="DataSeaarch" onChange={this.searchHandler} type="text" placeholder="Search" />
```
Comment (haishanh) [inline src/pages/Overall.js:82]: typo in `DataSeaarch`

File: src/pages/Overall.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        } else if (this.state.filterType === "voternumbers") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.voternumbers.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "update") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.update.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
```
Comment (haishanh) [inline src/pages/Overall.js:62]: Every block as this line `this.setState({ searchedRegions: searchResult })`, why not hoist it out?

File: src/pages/Overall.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        this.setState({ isDesktop: window.innerWidth > 750 });
+    }
+
+    changeHandler = (e) => {
+        const filter = e.target.value
+        this.setState({ filterType: filter })
+    }
+
+    searchHandler = (e) => {
+        var searchQuery = e.target.value
```
Comment (haishanh) [inline src/pages/Overall.js:35]: prefer `const` over `var` here

File: src/pages/Overall.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "voternumbers") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.voternumbers.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "update") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.update.toString().indexOf(searchQuery) !== -1
+            )
```
Comment (haishanh) [inline src/pages/Overall.js:61]: I think you can optimize this into either a "switch case" or use a "filterType to filter function" object here

**Comment by makara** [comment]
Thanks. Closing.

**Pull request closed without merging**